specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/StaticSites.json

```diff
@@ -0,0 +1,1703 @@
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/StringDictionary"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List secrets for a static site": {
+            "$ref": "./examples/ListStaticSiteSecrets.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "StaticSite": {
+      "description": "A static site.",
+      "type": "object",
+      "properties": {
+        "defaultHostname": {
+          "description": "The default autogenerated hostname for the static site.",
+          "type": "string"
+        },
+        "repositoryUrl": {
+          "description": "URL for the repository of the static site.",
+          "type": "string"
+        },
+        "branch": {
+          "description": "The target branch in the repository.",
+          "type": "string"
+        },
+        "customDomains": {
+          "description": "The custom domains associated with this static site.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "repositoryToken": {
+          "description": "A user's github repository token. This is used to setup the Github Actions workflow file and API secrets.",
+          "type": "string"
+        },
+        "buildProperties": {
+          "$ref": "#/definitions/StaticSiteBuildProperties",
+          "description": "Build properties to configure on the repository."
+        }
+      }
+    },
+    "StaticSiteARMResource": {
+      "description": "Static Site ARM resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/StaticSite",
+          "description": "Core resource properties",
+          "x-ms-client-flatten": true
+        },
+        "sku": {
+          "$ref": "./CommonDefinitions.json#/definitions/SkuDescription"
+        }
+      }
+    },
+    "StaticSiteBuildARMResource": {
+      "description": "Static Site Build ARM resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "StaticSiteBuildARMResource resource specific properties",
+          "properties": {
+            "buildId": {
```

Comment:
is this a readonly property which is only settable by the service and not by end customer

Reply:
correct, only our service can create these build objects and the corresponding id's associated with them. owners can list and delete them though